File: _includes/code/llms-txt/python/multi_tenancy.py
"""llms.txt snippet: multi-tenancy. Section "Python / TypeScript > Multi-tenancy"."""
import os
import weaviate
from weaviate.classes.init import Auth

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=os.environ["WEAVIATE_URL"],
    auth_credentials=Auth.api_key(os.environ["WEAVIATE_API_KEY"]),
)
client.collections.delete("Docs__MtPy")

# START llms_multi_tenancy
from weaviate.classes.config import Configure
from weaviate.classes.tenants import Tenant

client.collections.create(
    "Docs__MtPy",
    vector_config=Configure.Vectors.text2vec_weaviate(),
    multi_tenancy_config=Configure.multi_tenancy(enabled=True),
)
col = client.collections.use("Docs__MtPy")
col.tenants.create([Tenant(name="tenantA"), Tenant(name="tenantB")])
tenant_col = col.with_tenant("tenantA")
tenant_col.data.insert({"title": "Hello"})
res = tenant_col.query.hybrid("hello", limit=3)
# END llms_multi_tenancy

assert len(res.objects) == 1
assert len(col.tenants.get()) == 2
client.collections.delete("Docs__MtPy")
client.close()

File: _includes/code/llms-txt/typescript/multi_tenancy.ts
// llms.txt snippet: multi-tenancy. Section "Python / TypeScript > Multi-tenancy".
import weaviate, { vectors, configure } from 'weaviate-client';

const client = await weaviate.connectToWeaviateCloud(process.env.WEAVIATE_URL!, {
  authCredentials: new weaviate.ApiKey(process.env.WEAVIATE_API_KEY!),
});
await client.collections.delete('Docs__MtTs');

// START llms_multi_tenancy
await client.collections.create({
  name: 'Docs__MtTs',
  vectorizers: vectors.text2VecWeaviate(),
  multiTenancy: configure.multiTenancy({ enabled: true }),
});
const col = client.collections.use('Docs__MtTs');
await col.tenants.create([{ name: 'tenantA' }, { name: 'tenantB' }]);
const tenantCol = col.withTenant('tenantA');
await tenantCol.data.insert({ title: 'Hello' });
const res = await tenantCol.query.hybrid('hello', { limit: 3 });
// END llms_multi_tenancy

if (res.objects.length !== 1) throw new Error('expected 1 tenant object');
await client.collections.delete('Docs__MtTs');
await client.close();
